src/Microsoft.AspNetCore.Mvc.ViewFeatures/PageRemoteAttribute.cs

```diff
@@ -0,0 +1,90 @@
+
+using System;
+using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;
+using Microsoft.AspNetCore.Mvc.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Resources = Microsoft.AspNetCore.Mvc.ViewFeatures.Resources;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A <see cref="RemoteAttribute"/> for razor page handler which configures Unobtrusive validation 
+    /// to send an Ajax request to the web site. The invoked handler should return JSON indicating 
+    /// whether the value is valid.
+    /// </summary>
+    /// <remarks>Does no server-side validation of the final form submission.</remarks>
+    public class PageRemoteAttribute : RemoteAttribute
+    {
+        private readonly string _pageHandler;
+        private readonly string _pageName;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PageRemoteAttribute"/> class.
+        /// </summary>
+        /// <remarks>
+        /// Will use ambient page name and handler name when generating the URL where client
+        /// should send a validation request.
+        /// </remarks>
+        public PageRemoteAttribute()
+            : this(pageHandler: null)
+        {
+
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PageRemoteAttribute"/> class.
+        /// </summary>
+        /// <param name="pageHandler">
+        /// The handler name used when generating the URL where client should send a validation request.
+        /// </param>
+        public PageRemoteAttribute(string pageHandler)
+            : this(pageHandler, pageName: null)
+        { }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PageRemoteAttribute"/> class.
+        /// </summary>
+        /// <param name="pageHandler">
+        /// The page handler name used when generating the URL where client should send a validation request.
+        /// </param>
+        /// <param name="pageName">
+        /// The page name used when generating the URL where client should send a validation request.
+        /// </param>
+        /// <remarks>
+        /// <para>
+        /// If <paramref name="pageName"/> or <paramref name="pageHandler"/> is <c>null</c>, uses the corresponding
+        /// ambient value.
+        /// </para>
+        /// </remarks>
+        public PageRemoteAttribute(string pageHandler, string pageName)
+            : base()
+        {
+            _pageHandler = pageHandler;
+            _pageName = pageName;
+        }
+
+        /// <inheritdoc />
+        protected override string GetUrl(ClientModelValidationContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            
+            var services = context.ActionContext.HttpContext.RequestServices;
```

Comment:
I'd let @dougbu chime in, but would it be better to extract the "worker" part of `RemoteAttribute` in to it's own thing and using it rather than deriving from `RemoteAttribute` in this type?

Reply:
Hmm, this method duplicates most of the base method and only really changes the base's https://github.com/aspnet/Mvc/blob/a7301120b1971d6d047c63db5e6911aa134f99a2/src/Microsoft.AspNetCore.Mvc.ViewFeatures/RemoteAttribute.cs#L210-L214

One approach would be for `RemoteAttribute` to expose another `virtual method containing just that bit.

On the other hand, `RemoteAttribute`'s `RouteData` and `RouteName` bleed into this class unnecessarily and somewhat-confusingly. Thus I lean toward moving most of `RemoteAttribute` into a shared base and using that base here. (API Check might think the `RemoteAttribute` changes are breaking but we can work around that.)

(A "worker" `static` class is also possible but would make `RemoteAttribute` and `PageRemoteAttribute` more verbose. Sharing implementation through derivation feels fine to me here because the two attributes are so similar.)

Reply:
Thank you for the feedback. Just to clarify, the suggestion is to create a new RemoteBaseAttribute abstract class that contains all the "meat" with abstract GetUrl which each derived attribute has to implement?

Reply:
Call it `RemoteAttributeBase`. Otherwise, exactly right.

That said, I was incorrect about `RouteData` being irrelevant. Most `RemoteAttribute` properties *are* relevant, including `RouteData`. (`RouteName` is the only one that's specific to `RemoteAttribute`.) You're not using `RouteData` at the moment but should be i.e. call the `Page(string, string, object)` overload instead.